### ISSUE TITLE
Question with my workspace

### ISSUE THREAD

**Issue by star0w** [HyHuang1995/sp_orb_slam#1]
hello, I have met difficulties when using this code in my computer. Hope u can help!
Maybe this is related to gcc version or C++ 17? Could you share more information about the version of cmake and gcc? thx!:)

`EXAMPLE/CATKIN/WORK_SPACE/src/sp_orb_slam/orb_slam2/src/type/mappoint.cpp:177:44: error: no matching function for call to ‘std::shared_lock<std::shared_mutex>::shared_lock(std::shared_timed_mutex&)’
   shared_lock<shared_mutex> lock2(mMutexPos);
`
`error: no matching function for call to ‘std::shared_lock<std::shared_mutex>::shared_lock(std::shared_timed_mutex&)’
   shared_lock<shared_mutex> lock2(mMutexPos);
                                            ^
`
`/EXAMPLE/CATKIN/WORK_SPACE/src/sp_orb_slam/orb_slam2/include/orb_slam/viz/../type/keyframe.h:265:8: error: ‘shared_mutex’ in namespace ‘std’ does not name a type
   std::shared_mutex mMutexPose;
        ^~~~~~~~~~~~
`

**Comment by HyHuang1995** [comment]
Yes, it seems to me is related to gcc version. `shared_mutex` is available in g++-7 and newer version.

**Comment by star0w** [comment]
> Yes, it seems to me is related to gcc version. `shared_mutex` is available in g++-7 and newer version.

Sure.  I tried to update my gcc /g++ version from 5.8 to 7.5, however, it seems to be useless. Are there any other possible solutions for this?  :( THX!
